server/backend/database/mongo/client.go

```diff
@@ -1149,62 +1149,170 @@ func (c *Client) UpdateAndFindMinSyncedTicket(
 	), nil
 }
 
-// UpdateAndFindMinSyncedVersionVector updates the given serverSeq of the given client
-// and returns the SyncedVersionVector of the document.
-func (c *Client) UpdateAndFindMinSyncedVersionVector(
+// UpdateAndFindMinSyncedVersionVectorAfterPushPull returns min synced version vector
+func (c *Client) UpdateAndFindMinSyncedVersionVectorAfterPushPull(
 	ctx context.Context,
 	clientInfo *database.ClientInfo,
 	docRefKey types.DocRefKey,
-	serverSeq int64,
+	versionVector time.VersionVector,
 ) (time.VersionVector, error) {
-	// 01. update synced seq of the given client and document.
-	if err := c.UpdateSyncedSeq(ctx, clientInfo, docRefKey, serverSeq); err != nil {
-		return nil, err
-	}
+	var versionVectorInfos []database.VersionVectorInfo
+	var minVersionVector time.VersionVector
+	var clientVersionVector time.VersionVector
+	var filteredVersionVector time.VersionVector
 
-	// 02. find min synced seq of the given document.
-	result := c.collection(ColSyncedSeqs).FindOne(ctx, bson.M{
+	// 01. Find all version vector stored in db
+	cursor, err := c.collection(ColVersionVector).Find(ctx, bson.M{
 		"project_id": docRefKey.ProjectID,
 		"doc_id":     docRefKey.DocID,
-	}, options.FindOne().SetSort(bson.D{
-		{Key: "server_seq", Value: 1},
-	}))
-	if errors.Is(result.Err(), mongo.ErrNoDocuments) {
-		return nil, nil
+	})
+	defer cursor.Close(ctx)
```

Comment:
Do we need to call this function?

Reply:
I added it just in case we needed it. Is it okay not to have it?